UM/Settings/ContainerStack.py

```diff
@@ -232,7 +236,8 @@ def serialize(self):
 
         parser["metadata"] = {}
         for key, value in self._metadata.items():
-            parser["metadata"][key] = str(value)
+            if key not in self._metadata_serialization_excluded_keys:
```

Comment:
(non blocking) small taste difference: I would say
if key in self._metadata_serialization_excluded_keys:
  continue
<do all other things>

This prevents indenting a potentially big block. Also, I can wrap my head around it easier. Now you have to read: what does it mean to have a key that's not in the excluded list?